be/src/exprs/agg/approx_top_k.h

```diff
@@ -0,0 +1,466 @@
+            return;
+        }
+        this->data(state).is_init = true;
+        const auto kv = get_k_and_counter_num(ctx);
+        this->data(state).reset(kv.first, kv.second);
+    }
+
+    void process_null(FunctionContext* ctx, AggDataPtr __restrict state) const {
+        init_state_if_necessary(ctx, state);
+        this->data(state).process_null(1);
+    }
+
+    void merge(FunctionContext* ctx, const Column* column, AggDataPtr state, size_t row_num) const override {
+        DCHECK(column->is_binary());
+        Slice bytes = column->get(row_num).get_slice();
+
+        size_t start = 0;
+        int64_t null_count;
+        int32_t effective_counter_num;
+        // null count
+        std::memcpy(&null_count, bytes.data + start, sizeof(int64_t));
+        start += sizeof(int64_t);
+        // effective counter number
+        std::memcpy(&effective_counter_num, bytes.data + start, sizeof(int32_t));
+        start += sizeof(int32_t);
+
+        std::vector<typename ApproxTopKState<LT>::Counter> descrialize_counters;
+        descrialize_counters.reserve(effective_counter_num);
+        for (size_t i = 0; i < effective_counter_num; i++) {
+            descrialize_counters.emplace_back(i);
+        }
+
+        for (size_t i = 0; i < effective_counter_num; i++) {
+            // counter::value
+            if constexpr (IsSlice<CppType>) {
+                int64_t slice_size;
+                std::memcpy(&slice_size, bytes.data + start, sizeof(int64_t));
+                start += sizeof(int64_t);
+                uint8_t* slice_data = ctx->mem_pool()->allocate(slice_size);
+                std::memcpy(slice_data, bytes.data + start, slice_size);
+                start += slice_size;
+                descrialize_counters[i].value = Slice{slice_data, static_cast<size_t>(slice_size)};
+            } else {
+                std::memcpy(&descrialize_counters[i].value, bytes.data + start, sizeof(CppType));
+                start += sizeof(CppType);
+            }
+            // counter::count
+            std::memcpy(&descrialize_counters[i].count, bytes.data + start, sizeof(int64_t));
+            start += sizeof(int64_t);
+        }
+
+        init_state_if_necessary(ctx, state);
+        this->data(state).process_null(null_count);
+        this->data(state).merge(ctx->mem_pool(), descrialize_counters);
+    }
+
+    void update(FunctionContext* ctx, const Column** columns, AggDataPtr __restrict state,
+                size_t row_num) const override {
+        init_state_if_necessary(ctx, state);
+        const auto* column = down_cast<const InputColumnType*>(ColumnHelper::get_data_column(columns[0]));
+        const auto& value = AggDataTypeTraits<LT>::get_row_ref(*column, row_num);
+        this->data(state).template process<true>(ctx->mem_pool(), value, 1, false);
+    }
+
+    void serialize_to_column(FunctionContext* ctx, ConstAggDataPtr __restrict state, Column* to) const override {
+        DCHECK(to->is_binary());
```

Comment:
would it be nullable(binary?

Reply:
No, it won't be, defined by the definiation of the function.